### DISCUSSION TITLE
Is it possible to train a model on phonemes level and use it as streaming on device?

### DISCUSSION THREAD

**Discussion by make1986** [k2-fsa/sherpa-onnx#447] [Q&A]
Do you have an example of a pretrained model that I could try before I decide to invest time into it?

**Comment by csukuangfj** [comment]
You can find a list of pre-trained **streaming** models at
https://k2-fsa.github.io/sherpa/onnx/pretrained_models/online-transducer/index.html

However, none of them uses phonemes. They use BPE for English.

**Reply by ibsusu** [reply]
Has this changed since you started converting things with piper-phonemize?  Also, thanks for working on this for basically a year and then including espeak-ng -> piper-phonemize into this.  It's insane that this is so good and works on so many systems.

The amount of CMake craziness you've gone through is admirable.  I have no idea how you got so good at it.
